R/calc_dist_params.R

```diff
@@ -0,0 +1,159 @@
+  stopifnot(
+    "probability distribution params must be a named vector or NA" =
+      all(is.na(prob_dist_params)) ||
+      !is.null(names(prob_dist_params))
+  )
+
+  # extract mean and sd to see if conversion is possible
+  mean_sd <- c(
+    mean = summary_stats$central_tendency_spread$mean,
+    sd = summary_stats$central_tendency_spread$sd
+  )
+
+  # extract quantiles of distribution to calculate parameters as second choice
+  quantiles <- unlist(summary_stats$quantiles)
+  lower_quantiles <- quantiles[c("q_025", "q_05", "q_25", "q_50")]
+  upper_quantiles <- quantiles[c("q_75", "q_875", "q_95", "q_975")]
+
+  # extract median and range to calculate parameters as third choice
+  median_range <- c(
+    median = summary_stats$central_tendency_spread$median,
+    unlist(summary_stats$range)
+  )
+
+  if (!anyNA(mean_sd)) {
+    # make mean_sd a class with the name of the prob dist for multiple dispatch
+    prob_dist_params <- convert_params(
+      summary_stats = summary_stats,
+      prob_dist = prob_dist
+    )
+  } else if (!all(is.na(lower_quantiles)) && !all(is.na(upper_quantiles))) {
+    # use the first numeric quantile in the set to calculate the parameters
+    lower_percentile <- lower_quantiles[!is.na(lower_quantiles)][1]
+    upper_percentile <- upper_quantiles[!is.na(upper_quantiles)][1]
+
+    # extract the numeric quantiles from the vector names
+    percentiles <- as.numeric(c(
+        gsub(
+          pattern = "q_",
+          replacement = "",
+          x = names(lower_percentile),
+          fixed = TRUE
+        ),
+        gsub(
+          pattern = "q_",
+          replacement = "",
+          x = names(upper_percentile),
+          fixed = TRUE
+        )
+    ))
+
+    # TODO: make use of lognormal or lnorm consistent
+    if (prob_dist == "lognormal") prob_dist <- "lnorm"
+
+    # calculate the parameters from the percentiles
+    prob_dist_params <- extract_param(
+      type = "percentile",
+      values = c(lower_percentile, upper_percentile),
+      distribution = prob_dist,
+      percentiles = percentiles
+    )
+  } else if (!anyNA(median_range) && !is.na(sample_size)) {
+    prob_dist_params <- extract_param(
+      type = "range",
+      values = median_range,
+      distribution = prob_dist,
+      samples = sample_size
+    )
+  } else {
+    message(
+      "No adequate summary statistics available to calculate the parameters ",
+      "of the ", prob_dist, " distribution"
+    )
+  }
+
+  # return params
+  prob_dist_params
+}
+
+#' Converts summary statistics to parameters of a probability distribution
+#'
+#' @description The summary_stats object needs to have a class attribute with
+#' the name of the probability distribution in order for the correct conversion
+#' function to be dispatched.
+#'
+#' @inheritParams new_epidist
+#' @inheritParams epidist
+#'
+#' @return Named vector
+#' @keywords internal
+convert_params <- function(summary_stats,
```

Comment:
Might be nice in the future to extend to more conversions, e.g. could also estimate lognormal parameters using the median instead of the mean.

Reply:
I will continue to add conversions functions as we develop. For now I've opened an issue (#98), and suggested a new way to format the conversion functions. Let me know what you think.